### ISSUE TITLE
Getting error code in Lambda and other issues

### ISSUE THREAD

**Issue by rlink23** [bklavet/Echo-skill-to-control-Directv#53]
So I am running the test in Lambda and it is failing. Then I tried to run test in Alexa app and it says The "remote endpoint could not be called, or the response it returned was invalid". I am trying to just run the main DVR in the living room, and I have port forwarding set up on the direct TV ip address. i also set up DMZ for that same IP. I am not sure what I am doing incorrect at this point.. Please help!! I am a beginner but I follow directions well So I think I have done everything correctly.

**Comment by rlink23** [comment]
Also I am not sure where to input my Mac address.. I did a different tutorial and got it all to wok but at the end Alexa did not understand what I was saying. This seems to be the better tutorial but cannot figure it out for the life of me... :/

**Comment by bklavet** [comment]
hello rlink23!  First thing is to make sure you can control your receiver from your lan and then from outside your network.
check here and run through the list.  if anything seems confusing let me know and I will try to clarify.
https://github.com/bklavet/Echo-skill-to-control-Directv/issues/12

if this is the main genie you wont need to enter anything for the mac.  The mac is needed for additional receivers.
I am guessing a little but make sure you are entering your wan ip address in the index file you are uploading to lambda and not your local ip address.

**Comment by rlink23** [comment]
I tried that. Seems like it wasn't working. Do I use the ip address of the receiver or do I do use my main public ip address?



Sent from Outlook<http://aka.ms/weboutlook>


________________________________

**Comment by bklavet** [comment]
you will want to use the public ip address.

**Comment by rlink23** [comment]
Ok I will try it when I get home... It also seemed like when I was adding the zip file it didn't seem like it was working. Lambda site It didn't like it and was throwing an error. Should it  I know I can put the coding inline on Lambda but I am not sure where to insert the Alexa skill code... Is that even possible and not doing the zip file?


Sent from Outlook<http://aka.ms/weboutlook>


________________________________

**Comment by bklavet** [comment]
when you add the zip file you will get a response saying something along the lines that it is a zipped file and can not be edited inline.  That is normal, and something else to remember for future if your public ip is not static and changes when you reset your modem or something, you will want to update the index file with the new public ip, zipp it up and upload to your function.

if you know what your public ip is and you have your forwarding set up you should be able to enter this into a computer/phone that is not on the same network as your directv and get the main receiver to pause
http://[wan_ip]:8080/remote/processKey?key=pause&hold=keyPress

**Comment by rlink23** [comment]
ok, I will try it.. I will be in touch if I can't get it to work.. The only other thing i can think of is my receiver is different. I run the HR54  (genie).


And you are saying if I test in Lambda it is normal that it won't test properly?


I know I am being a pain in the rear haha


Sent from Outlook<http://aka.ms/weboutlook>


________________________________

**Comment by rlink23** [comment]
Also what Node should I be runnin? node.Js 4.3?


Sent from Outlook<http://aka.ms/weboutlook>
________________________________

**Comment by bklavet** [comment]
ok, I used node 4.3 but 6.10 would probably work too.  if you want to test through lambda, ( i am assuming you can control your receiver from outside your network from the browser.  If you can't there is no way your lambda function will work)  Go to your alexa developer portal and click on the test tab, go to service simulator and type in "pause"  copy what shows in the box labled service request and go to lambda and configure a test event and paste the service request there and you can test directly in lambda.

One more time though, you need to be able to control your receiver from outside your network, if you can't do that there is zero chance the skill will work.

**Comment by rlink23** [comment]
Well I'm not sure exactly what I did. But I got it to work. I think it was having trouble reading the zip file. Thank you so much for your work coding and doing the real work so I could make this work. I am a beginner at coding but slowly am learning the verbage and all of that. And its all self taught. I actually enjoy it even tho it makes me pull my hair out sometimes.  Lol



Sent from my T-Mobile 4G LTE Device


-------- Original message --------
